### PR TITLE
Add environment variable to configure MTU

### LINKED ISSUE #356: CNI doesn't support configuring MTU
As far as I understand the maximum transmission unit for each POD is hard coded to 9001 after the completion of #167. This currently causes problems since paths to some hosts our applications need to access has a much lower MTU.

What happens is typically that TCP connections hang waiting for response packages that are to big to be delivered since the agreed on maximum segment size (MSS) of the connection is too high.

This would not happen if the network paths to these hosts where letting through ICMP “Destination Unreachable, fragmentation required, and DF flag set“ packets, but alas, they don't.

So I request a way to configure the MTU to something lower than 9001.

### PR DESCRIPTION
**Pull request by mogren**
*Issue #, if available:* Resolves #356

*Description of changes:*
* Added `AWS_VPC_ENI_MTU` environment variable to set MTU
* Added unit tests to verify limits.
* Corrected some comments

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/networkutils/network.go [hunk 1]
```diff
@@ -929,3 +935,27 @@ func (n *linuxNetwork) UpdateRuleListBySrc(ruleList []netlink.Rule, src net.IPNe
...
+		mtu, err := strconv.Atoi(envMTUValue)
+		if err != nil {
+			log.Error("Failed to parse "+envMTU+"; will use ", maximumMTU, err.Error())
+			return maximumMTU
+		}
+		// Restrict range between jumbo frame and the maximum required size to assemble.
+		// Details in https://tools.ietf.org/html/rfc879 and
+		// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html
+		if mtu < minimumMTU {
+			log.Error(""+envMTU+" too low; will use ", minimumMTU)
```
Comment (shyamjvs) [inline pkg/networkutils/network.go]: Why are you appending empty string in front of envMTU?

Comment (mogren) [reply pkg/networkutils/network.go]: That was just sloppy, copied from the previous error log line. I've updated to use `log.Errorf()`

**Comment by shyamjvs** [comment]
/lgtm

Nice!

File: README.md [hunk 2]
```diff
@@ -152,6 +152,16 @@ To select an `ENIConfig` based upon availability zone set this to `failure-domai
 
 ---
 
+`AWS_VPC_ENI_MTU`
```
Comment (vsiddharth) [inline README.md]: Is this option cluster scoped against node scope?

File: pkg/networkutils/network.go [hunk 3]
```diff
@@ -77,18 +77,22 @@ const (
...
-	// defaultConnmark is the default value for the connmark described above.  Note: the mark space is a little crowded,
+	// defaultConnmark is the default value for the connmark described above. Note: the mark space is a little crowded,
 	// - kube-proxy uses 0x0000c000
 	// - Calico uses 0xffff0000.
 	defaultConnmark = 0x80
 
-	// MTU of ENI - veth MTU defined in plugins/routed-eni/driver/driver.go
-	ethernetMTU = 9001
+	// envMTU gives a way to configure the MTU size for new ENIs attached. Range is from 576 to 9001.
+	envMTU = "AWS_VPC_ENI_MTU"
```
[the remark this answers is not in the document]
Comment (mogren) [reply pkg/networkutils/network.go]: No, they will all get the same setting to keep things simple.

File: README.md (hunk 2 above)
Comment (mogren) [reply README.md]: All nodes in the cluster will get the same setting.

**Pull request merged**